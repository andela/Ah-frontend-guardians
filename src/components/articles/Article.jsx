import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getSingleDataThunk } from '../../actions/ArticleActionCreator';
import { NavBar } from '../navBar/NavBar';
import Footer from '../Footer/Footer';
import renderHTML from 'react-render-html';

export class Article extends Component {
  componentDidMount() {
    if (this.props.match) {
      const slug = this.props.match.params.slug;

      this.props.getSingleDataThunk(slug);
      const {
        article: { body }
      } = this.props;
    }
  }

  render() {
    if (this.props.articleReducer.article === undefined) return <div />;
    const {
      article: {
        author,
        title,
        body,
        description,
        read_time,
        slug,
        created_at,
        updated_at,
        tags,
        image_url
      }
    } = this.props.articleReducer;

    return (
      <div>
        <NavBar {...this.props} />
        <div id="singleArticle">
          <div id="authorColor">{author && author.username}</div>

          <div id="articleReadTime">{read_time + ' min read'}</div>
          <div id="contentA">
            <div>
              <h2>{title}</h2>
            </div>
            <div>
              <h5>{description}</h5>
            </div>
            <div id="bodycolor">{renderHTML(body)}</div> <br />
            
          </div><br/>
          <div id="tagstyle" className="btn btn-primary">{tags}</div>
        </div>
        <Footer />
      </div>
    );
  }
}
Article.propTypes = {
  article: PropTypes.shape({
    slug: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string,
    body: PropTypes.string,
    created_at: PropTypes.string,
    updated_at: PropTypes.string,
    image_url: PropTypes.string,
    author: PropTypes.shape({
      username: PropTypes.string,
      email: PropTypes.string
    })
  })
};
Article.defaultProps = {
  article: {
    slug: '',
    title: '',
    description: '',
    body: '',
    created_at: '',
    updated_at: '',
    image_url: null,
    author: { username: '', email: '' }
  }
};
export const mapStateToProps = state => state;
export default connect(
  mapStateToProps,
  { getSingleDataThunk }
)(Article);
