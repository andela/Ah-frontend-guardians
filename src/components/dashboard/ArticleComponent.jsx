import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { NavLink } from 'react-router-dom';
import default_image from '../../images/default_image.jpeg';

export const Article = (props) => {
  const getPlainText = data => data.replace(/<(?:.|\n)*?>/gm, '');
  const slug = props.article && props.article.slug;
  const url = 'article/' + slug;

  return (
    <NavLink to={url} id="Nav">
      <div className="article-box welocme_msg">
        <div>
          {' '}
          <img id="articleImage" src={default_image} width="300" height="150" />
        </div>
        <div id="cont">
          <h3>{props.article && props.article.title}</h3>
          <p className="text-dark">
            {props.article && props.article.description.substring(0, 50) + '...'}
          </p>
          <div>
            by <font color="red">{props.article && props.article.author.username}</font>
            &nbsp;| &nbsp;
            <font style={{ color: `${'#25ABD1'}` }}>
              {props.article && props.article.read_time + ' min read'}
            </font>
          </div>
        </div>
      </div>
    </NavLink>
  );
};
Article.propTypes = {
  article: PropTypes.shape({
    author: PropTypes.object,
    body: PropTypes.string,
    title: PropTypes.string,
    created_at: PropTypes.string
  })
};
export default Article;
