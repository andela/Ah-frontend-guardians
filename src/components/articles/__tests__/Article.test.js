import React, { Component } from 'react';
import { shallow } from 'enzyme';
import ArticleComponent, { Article } from '../Article';
import { mapStateToProps } from '../Article'

const props = {
  articleReducer: {
    article: {},
  },
}

describe('Fetch Article', () => {
  let wrapper;

  const mockFetch = jest.fn();

  afterEach(() => {
    mockFetch.mock.calls = [];
  });
  it('should render without crushing', () => {
    const wrapper = shallow(<Article {...props}/>)

    expect(wrapper.find('h3').length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render without crashing', () => {
    const wrapper = shallow(<Article />)

    expect(wrapper).toMatchSnapshot()
  })
  it('should return article state', () => {
    const wrapper = shallow(
      <Article {...props}
        articleReducer={{}}
        match={{ params: { slug: 'asdfghjkl' } }}
        getSingleDataThunk={jest.fn()}
      />,
    );

    wrapper.setProps({
      articleReducer: {
        article: {
          body: "body",
        },
      },
    })
    expect(wrapper.instance().props.articleReducer).toEqual({
        article: {
          body: "body",
        },
    })
  });
  it('mapStateToProps should return the right value', () => {
    const mockedState = {
          title: "fahad",
    };
    const state = mapStateToProps(mockedState);

    expect(state.title).toEqual('fahad');
});
});
