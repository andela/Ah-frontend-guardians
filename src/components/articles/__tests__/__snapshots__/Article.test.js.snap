// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch Article should have an h2 element 1`] = `ShallowWrapper {}`;

exports[`Fetch Article should render without crashing 1`] = `ShallowWrapper {}`;
